src/8a860_len_3f30.c

```diff
@@ -409,6 +420,127 @@ s32 popup_menu_update(void) {
+                    gPopupMenu->unk_330 = MSG_Menus_Items;
+                    break;
+                case POPUP_MENU_SWITCH_PARTNER:
+                    var_s6 = 139;
+                    var_s5 = 139;
+                    gPopupMenu->unk_330 = MSG_Menus_PartyMember;
+                    gPopupMenu->unk_338 = 1;
+                    gPopupMenu->unk_33C = MSG_Menus_0063;
+                    break;
+                case POPUP_MENU_TRADE_FOR_BADGE:
+                    var_s6 = 162;
+                    var_s5 = 140;
+                    gPopupMenu->unk_340 = -16;
+                    gPopupMenu->unk_330 = MSG_MenuTip_0032;
+                    gPopupMenu->unk_33C = MSG_Menus_0070;
+                    break;
+                case POPUP_MENU_UPGRADE_PARTNER:
+                    var_s6 = 146;
+                    var_s5 = 140;
+                    gPopupMenu->unk_340 = -8;
+                    gPopupMenu->unk_330 = MSG_Menus_PartyMember;
+                    gPopupMenu->unk_338 = 1;
+                    gPopupMenu->unk_33C = MSG_Menus_006A;
+                    break;
+                case POPUP_MENU_SELL_ITEM:
+                    var_s6 = 167;
+                    var_s5 = 145;
+                    gPopupMenu->unk_340 = -24;
+                    gPopupMenu->unk_330 = MSG_Menus_Items;
+                    gPopupMenu->unk_33C = MSG_Menus_0065;
+                    break;
+                case POPUP_MENU_READ_LETTER:
+                    var_s6 = 170;
+                    var_s5 = 170;
+                    gPopupMenu->unk_340 = -32;
+                    gPopupMenu->unk_330 = MSG_MenuTip_0033;
+                    gPopupMenu->unk_33C = MSG_Menus_006D;
+                    break;
+                case POPUP_MENU_TAKE_FROM_CHEST:
+                    var_s6 = 145;
+                    var_s5 = 145;
+                    gPopupMenu->unk_330 = MSG_Menus_00D7;
+                    gPopupMenu->unk_33C = MSG_Menus_00D6;
+                    break;
+                case POPUP_MENU_READ_DIARY_PAGE:
+                    var_s6 = 140;
+                    var_s5 = 140;
+                    gPopupMenu->unk_340 = -16;
+                    gPopupMenu->unk_330 = MSG_Menus_00CE;
+                    gPopupMenu->unk_33C = MSG_Menus_006C;
+                    break;
+                case POPUP_MENU_READ_POSTCARD:
+                    var_s6 = 149;
+                    var_s5 = 149;
+                    gPopupMenu->unk_340 = -16;
+                    gPopupMenu->unk_330 = MSG_Menus_00D0;
+                    gPopupMenu->unk_33C = MSG_Menus_006D;
+                    break;
+                case POPUP_MENU_POST_OFFICE:
+                    var_s6 = 131;
+                    var_s5 = 131;
+                    gPopupMenu->unk_330 = MSG_Menus_PartyMember;
+                    gPopupMenu->unk_338 = 1;
+                    gPopupMenu->unk_33C = MSG_Menus_006F;
+                    break;
+            }
+            msgWidth = get_msg_width(gPopupMenu->unk_330, 0);
```

Comment:
Figured out what this code does:
![image](https://github.com/pmret/papermario/assets/737603/9305c4ad-c26b-4ed0-9e3a-87ede5a6d32f)

It calculates the size of the title textbox by taking `(var_s5 - textWidth) * 40%`. Afterwards, unk_334 is filled with the x offset of the title text box compares to the rest of the menu.

I don't really understand why `var_s5` and `var_s6` differ; it causes some titles to be off-center. 
![image](https://github.com/pmret/papermario/assets/737603/23a29fff-8575-4a1b-998d-d1e675b8159f)